File: backend/src/main/java/org/ilisi/backend/model/Education.java
package org.ilisi.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.YearMonth;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Education {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(updatable = false, nullable = false)
    private String id;
    private String title;
    private String studyField;
    private String description;
    private YearMonth startAt;
    private YearMonth endAt;
    private String location;

    @ManyToOne(cascade = CascadeType.MERGE)
    private Institut institut;
}

File: frontend/src/app/models/education.ts
class Education {
    id?:string;
    title: string;
    studyField: string;
    description: string;
    startAt: string;
    endAt: string;
    institut?: any;
    constructor(
        id: string,
        title: string,
        studyField: string,
        description: string,
        startAt: string,
        endAt: string,
        institut: any
    ) {
        this.id = id;
        this.title = title;
        this.studyField = studyField;
        this.description = description;
        this.startAt = startAt;
        this.endAt = endAt;
        this.institut = institut;
    }
}

export default Education;
